application-collabora-ui/src/main/resources/Collabora/Code/Main.xml

```diff
@@ -203,35 +196,34 @@
     </property>
     <property>
       <code>require(['jquery'], function($) {
+  const collaboraPath ='/rest/collabora/files/';
   $(function() {
-    const server = $("#collaboraServer").data('collaboraServer');
     const fileId = $("#collaboraServer").data('fileId');
-    const userCanWrite = $("#collaboraServer").data('mode') == 'edit';
-    const wopiResourceURL = window.location.origin + '/xwiki/rest/wopi/files/' + fileId
-      + '?userCanWrite=' + userCanWrite;
-    const discoveryURL = XWiki.contextPath + '/rest/discovery';
-    $.getJSON(discoveryURL, $.param({
-      xpage: 'plain',
-      outputSyntax: 'plain',
-      server: server,
-      ext: fileId.split('.').pop(),
+    const userCanWrite = $("#collaboraServer").data('mode') === 'edit';
+    const errorMessage = $("#collaboraServer").data('error');
+    const wopiResourceURL = window.location.origin + '/xwiki' + collaboraPath + encodeURIComponent(fileId);
+    const tokenURL = XWiki.contextPath + collaboraPath + encodeURIComponent(fileId) + '/token';
+    $.getJSON(tokenURL, $.param({
       fileId: fileId
```

Comment:
You pass the file id in the path so I doubt you still need it in the query string.